devnotes/tpm.md

```diff
@@ -0,0 +1,228 @@
+This document is an attempt to gather information about the issues faced with
+TPMs on POWER9 and what are the potential solutions to them.
+
+## Some definitions
+
+* fTPM - Firmware-based TPM implementation
+
+## Problems
+
+### POWER9 cannot generate LPC TPM cycles
+
+This is a hardware limitation and cannot be worked around in firmware.
+
+The consequence is that regular LPC TPM cannot be used on POWER9 by directly
+connecting it to the TPM LPC header (one might wonder why have the LPC TPM
+header in the first place...).
+
+## Quick overview
+
+| Aspect\Approach | [I2C]        | [On BMC] | [FPGA LPC] | [FPGA SPI] | [On MCU]      | [LibreBMC]    | [SBE]    |
+| --------------- | -----        | -------- | ---------- | ---------- | --------      | ----------    | -----    |
+| Interface       | I2C          | -        | LPC        | SPI        | I2C           | ?             | -        |
```

Comment:
Let's list available buses: I2C, LPC, SPI, anything else is feasible? PCI/PCIe I'm nor sure how LibreBMC is connected?

Reply:
Bus is constrained by an approach and it's basically one-to-one correspondence.

Reply:
So this seems to be redundant information and line Approach could be used for meaningful naming with consistent pattern. Now in Approach we have either bus, either MCU/CPU/FPGA, either MCU/CPU+bus name, which is inconsistent and redundant. What about explaining but in interface field and leave Approach with the naming proposed in my other comment?

Reply:
Ok, I see you already used approach with fTPM and dTPM, so I think this is solved.

Reply:
Naming columns according to interfaces will create identically named columns. I see your point on naming approaches and tried to make them more consistent. Previous naming was meant to just be short so that table doesn't get too wide.

Reply:
Agree that naming according to interfaces doesn't make sense. New naming is way better IMO.